bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeApi.java

```diff
@@ -45,33 +48,56 @@
 @NonNullByDefault
 public class SenecHomeApi {
     private static final String HTTP_PROTO_PREFIX = "http://";
+    private static final String HTTPS_PROTO_PREFIX = "https://";
 
     private final Logger logger = LoggerFactory.getLogger(SenecHomeApi.class);
     private final HttpClient httpClient;
     private final Gson gson = new Gson();
     private String hostname = "";
 
     public SenecHomeApi(HttpClient httpClient) {
-        this.httpClient = httpClient;
+        this.httpClient = configureHttpClient(httpClient);
     }
 
     public void setHostname(String hostname) {
         this.hostname = hostname;
     }
 
+    private HttpClient configureHttpClient(HttpClient httpClient) {
+        // Configure SSL context factory to trust all certificates
+        SslContextFactory.Client sslContextFactory = new SslContextFactory.Client();
+        sslContextFactory.setTrustAll(true);
+
+        // Apply SSL context factory to the HttpClient builder
+        httpClient = new HttpClient(sslContextFactory);
+
+        // Start the HttpClient
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            logger.error("Issue with starting the http client", e);
+        }
+
+        return httpClient;
+    }
+
```

Comment:
This was provided by ChatGPT to allow connecting to https devices with self-signed certificates.